Lambdas for Remaining Modules JPA -> ZK

garyrussell opened the pull request.

garyrussell committed:
Lambdas for Remaining Modules JPA -> ZK

artembilan commented on spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java:
```diff
@@ -703,55 +577,35 @@ public void testNoInitialIdleDelayWhenRecentNotSupported() throws Exception {
-				return null;
-			}
+		doAnswer(invocation -> {
+			Thread.sleep(5000);
```
Can we get rid of this long sleep somehow?
I understand that this is a `LongRunningIntegrationTest`, but still...
Sometime I run `testAll` locally and our CI plan for IO performs `testAll`, too. Twice.

garyrussell replied:
I can't get too excited about this one test, at 5 seconds, it represents only 6% of the total run time for this class. Feel free to put a JIRA on the backlog to look at this 80 second test class.

artembilan replied:
Here you are : https://jira.spring.io/browse/INT-4150

artembilan commented on spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java:
```diff
@@ -898,34 +720,26 @@ public void testImapLifecycleForRaceCondition() throws Exception {
+				}
+				catch (javax.mail.MessagingException e) {
+					if (e.getCause() instanceof NullPointerException) {
+						e.printStackTrace();
```
No `printStackTrace()`.

artembilan commented on spring-integration-mail/src/test/java/org/springframework/integration/mail/ImapMailReceiverTests.java:
```diff
@@ -898,34 +720,26 @@ public void testImapLifecycleForRaceCondition() throws Exception {
+				}
+				catch (Exception ignore) {
+					// ignore
+					ignore.printStackTrace();
```
DITTO

artembilan commented on spring-integration-redis/src/test/java/org/springframework/integration/redis/util/AggregatorWithRedisLocksTests.java:
```diff
@@ -131,17 +131,13 @@ public void testLockThreeGroups() throws Exception {
+				in2.send(new GenericMessage<String>("bar", stubHeaders(2, 2, 1)));
+			}
+			catch (Exception e) {
+				e.printStackTrace();
```
Again

artembilan commented on spring-integration-xmpp/src/test/java/org/springframework/integration/xmpp/inbound/ChatMessageListeningEndpointTests.java:
```diff
@@ -146,15 +133,9 @@ public void testWithErrorChannel() throws NotConnectedException {
-		});
+		outChannel.subscribe(message -> {
+throw new RuntimeException("ooops");
+});
```
Something wrong with formatting.

artembilan reviewed (changes requested):
The slow email tests should be reconsidered in the separate task. Here I would like do not see `printStackTrace()` and that simple code formatting issue.
Thanks

garyrussell committed:
Polishing - PR Comments and Closeable Warnings

Eclipse emits bogus warnings with exceptions in lambdas.
Even though the lambda might run on another thread, elipse thinks it could
cause the context to not be closed.

garyrussell committed:
SPR-14854: MessageChannel is now a @FunctionalInterface

artembilan commented:
Merged as https://github.com/spring-projects/spring-integration/commit/67d6cd0c89ecb6e948887fd25de6f86705eac0e9
